tests/common.py

```diff
@@ -170,19 +170,26 @@ def wait_for_dataset_version_export_success(stub, metadata, dataset_id, dataset_
             metadata=metadata,
         )
         raise_on_failure(response)
-        export = response.dataset_version.export_info.clarifai_data_protobuf
-        if export.status.code == status_code_pb2.DATASET_VERSION_EXPORT_SUCCESS:
-            break
-        elif export.status.code in (
-            status_code_pb2.DATASET_VERSION_EXPORT_PENDING,
-            status_code_pb2.DATASET_VERSION_EXPORT_IN_PROGRESS,
-        ):
-            time.sleep(1)
+
+        for field in ["clarifai_data_protobuf", "clarifai_data_json"]:
+            if not response.dataset_version.export_info.HasField(field):
+                continue
```

Comment:
that's true. i understand what you're trying to achieve: make this method work with any exports.

there are two things that are not actual issues in current code but could be in the future:
* we *know* that `PutDatasetVersionExports` does set the export to `PENDING`, but the test does not assert that status is `PENDING` before calling `wait_for_dataset_version_export_success`.
* an extra side-effect is: calling `PostDatasetVersions` and then `wait_for_dataset_version_export_success` is going to continuously call `GetDatasetVersion` without any `time.sleep`. 

in order to makes the test method's intention clearer,  should we add the `["clarifai_data_protobuf", "clarifai_data_json"]` as parameter for `wait_for_dataset_version_export_success`? in this way, we avoid the two above-mentioned issues and we are still able to make this method work with any exports (controlled by parameter).

Reply:
I don't understand the second issue. If no exports are set, then it will call `GetDatasetVersion` once, `continue` for all fields of the export, and fall into the `else` block where it breaks and returns. The only way it will call `GetDatasetVersion` again is if we break the `for`-loop which is only done after a sleep. I think I am missing something.

As for the suggestion, I agree. Let's set the desired export types as parameters so that it is explicit what the test actually wants to wait for.

Reply:
> I don't understand the second issue...

ignore the second issue. i did mis-read the code.

> As for the suggestion, I agree. Let's set the desired export types as parameters so that it is explicit what the test actually wants to wait for.

great! now that we have the expected export fields as parameters, we could additionally raise error when field is not set. the test does know for sure that field is expected to be set.

Reply:
Oops, rushed this change and forgot about that. Thanks for pointing it out.